### ISSUE TITLE
Time is faster for unfocused windows

### ISSUE THREAD

**Issue by cuppachino** [bevyengine/bevy#13070]
## Bevy version

0.13.2

## \[Optional\] Relevant system information

```rust
AdapterInfo { name: "NVIDIA GeForce RTX 2060", vendor: 4318, device: 7944, device_type: DiscreteGpu, driver: "NVIDIA", driver_info: "551.86", backend: Vulkan }
```

## What you did

1. Run an example with an inspector plugin.
2. Click on the window. 
3. Note `Time<Virtual>`'s `elapsed` and `delta`.
4. Click out of the window. 
5. `elapsed` accumulates faster and `delta` is erratic.

## What went wrong

I expected the virtual timer to tick at the same speed regardless of window focus state.

## Additional information

![time_too_fast](https://github.com/bevyengine/bevy/assets/92656523/b339a7ff-7f89-4f3e-8d1e-015879e3c005)

**Comment by Earthmark** [comment]
I tried to replicate this locally, here's my findings!

@cuppachino , where is the `LogicStep` type from?

And I'm curious, is your bug report that the timer in the `LogicStep` type advanced more quickly than you intended, or that when a window is unfocused the update rate has different characteristics?

## Experiment Setup

The example program I used
```toml
[package]
name = "time_test"
version = "0.1.0"
edition = "2021"

[dependencies]
bevy="*"
bevy-inspector-egui = "*"

[profile.dev]
opt-level = 1

[profile.dev.package."*"]
opt-level = 3
```

```rust
use bevy::{input::common_conditions::input_toggle_active, prelude::*};

fn main() {
    App::new()
        .add_plugins(DefaultPlugins.set(WindowPlugin {
            primary_window: Some(Window {
                //present_mode: bevy::window::PresentMode::Immediate,
                ..default()
            }),
            ..default()
        }))
        .add_plugins(
            bevy_inspector_egui::quick::WorldInspectorPlugin::default()
                .run_if(input_toggle_active(true, KeyCode::Escape)),
        )
        .run();
}
```

## My findings

### Virtual Time (TLDR: Appears WAI)

I did find `Virtual` and `Real` time elapsed at slightly different rates, but it was minimally different, less than a second over a hundred seconds.

`Virtual` time seems consistent based on the elapsed time of the previous update tick, and adapted based on the update loop time. For me it seems it advances roughly close to real time, besides jump discontinuities when I drag the window where real time advances, but virtual time is paused during the drag.

### Fixed time (TLDR: Appears WAI)

I adjusted my Fixed timestep to step to 0.5 seconds, and it advanced at a consistent rate (although jumping every 0.5 seconds) focused or unfocused, relative to `Virtual` time. It appears at least on my machine to be fairly consistent.

### LogicStep time (The problem banana)

I do not see this type in Bevy's libraries, but this appears to be the type that is having the rate issue.

Was this a name of a previously removed type?

**Comment by cuppachino** [comment]
@Earthmark `LogicStep` is essentially a duplicate of `Fixed`. My mistake was running
